### PR TITLE
Remove use of deprecated dart::common::make_unique

### PR DESCRIPTION
**Pull request by jslee02**
***

**Before creating a pull request**

- [x] Document new methods and classes
- [x] Format code with `make format`

**Before merging a pull request**

- [ ] Set version target by selecting a milestone on the right side
- [ ] Summarize this change in `CHANGELOG.md`
- [ ] Add unit test(s) for this change

### CODE REVIEW

**Comment by jslee02** [comment]
It seems two build tests are failing:
- The macOS build fails due to failure on installing dependencies.
- The documentation build fails for unknown issue.

I'd move the failing builds to the allowed failures list for now and fix them later.

File: .ci/script_cmake.sh [hunk 1]
```diff
@@ -8,4 +8,4 @@ mkdir build
 cd build
 cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DTREAT_WARNINGS_AS_ERRORS=ON ..
 make -j4 tests
-make test
+ctest
```
Comment (brianhou) [inline .ci/script_cmake.sh]: Why?

Comment (jslee02) [reply .ci/script_cmake.sh]: Just to see if it makes any difference that turned out not. Feel free to revert it.

**Comment by jslee02** [comment]
Yes, exactly!

**Pull request merged**